capplet/msm-desktop-app-dialog.c

```diff
@@ -0,0 +1,311 @@
+    gtk_window_set_position (GTK_WINDOW (dialog), GTK_WIN_POS_CENTER);
+    gtk_widget_set_size_request (GTK_WIDGET (dialog), 600, 400);
+    gtk_window_set_modal (GTK_WINDOW (dialog), TRUE);
+
+    dialog->entry = gtk_search_entry_new ();
+    g_signal_connect (dialog->entry,
+                     "search-changed",
+                      G_CALLBACK (on_search_entry_changed),
+                      dialog);
+
+    gtk_entry_set_placeholder_text (GTK_ENTRY (dialog->entry), _("Search Applications…"));
+    gtk_entry_set_width_chars (GTK_ENTRY (dialog->entry), 30);
+    gtk_entry_set_activates_default (GTK_ENTRY (dialog->entry), TRUE);
+    gtk_entry_set_input_hints (GTK_ENTRY (dialog->entry), GTK_INPUT_HINT_NO_EMOJI);
+
+    searchbar = gtk_search_bar_new ();
+    gtk_container_add (GTK_CONTAINER (searchbar), dialog->entry);
+    gtk_widget_set_hexpand (searchbar, TRUE);
+    gtk_box_pack_start (GTK_BOX (content_area), searchbar, FALSE, FALSE, 0);
+
+    dialog->listbox = gtk_list_box_new ();
+    g_object_set (dialog->listbox, "margin", 6, NULL);
+    gtk_list_box_set_activate_on_single_click (GTK_LIST_BOX (dialog->listbox), FALSE);
+    gtk_list_box_set_sort_func (GTK_LIST_BOX (dialog->listbox), list_sort_app_func, dialog, NULL);
+    gtk_list_box_set_header_func (GTK_LIST_BOX (dialog->listbox), list_header_app_func, dialog, NULL);
+    gtk_list_box_set_filter_func (GTK_LIST_BOX (dialog->listbox), list_filter_app_func, dialog, NULL);
+    g_signal_connect (dialog->listbox,
+                     "row-selected",
+                      G_CALLBACK (on_row_selected),
+                      dialog);
+
+    gtk_dialog_add_button (GTK_DIALOG (dialog), _("_Close"), GTK_RESPONSE_CANCEL);
+    gtk_dialog_add_button (GTK_DIALOG (dialog), _("_Add"), GTK_RESPONSE_OK);
+    gtk_dialog_set_default_response (GTK_DIALOG (dialog), GTK_RESPONSE_OK);
+
+    sw = gtk_scrolled_window_new (NULL, NULL);
+    gtk_scrolled_window_set_policy (GTK_SCROLLED_WINDOW (sw),
+                                    GTK_POLICY_NEVER,
+                                    GTK_POLICY_AUTOMATIC);
+    gtk_container_add (GTK_CONTAINER (sw), dialog->listbox);
+    gtk_box_pack_start (GTK_BOX (content_area), sw, TRUE, TRUE, 0);
+
+    searchbutton = gtk_toggle_button_new ();
+    gtk_widget_set_valign (searchbutton, GTK_ALIGN_CENTER);
+    searchimage = gtk_image_new_from_icon_name ("edit-find-symbolic", GTK_ICON_SIZE_MENU);
+    gtk_container_add (GTK_CONTAINER (searchbutton), searchimage);
+
+    context = gtk_widget_get_style_context (searchbutton);
+    gtk_style_context_add_class (context, "image-button");
+    gtk_style_context_remove_class (context, "text-button");
+
+    headerbar = gtk_dialog_get_header_bar (GTK_DIALOG (dialog));
+    gtk_header_bar_pack_end (GTK_HEADER_BAR (headerbar), searchbutton);
+    g_object_bind_property (searchbutton, "active", searchbar, "search-mode-enabled", G_BINDING_BIDIRECTIONAL);
+}
+static GObject *
```

Comment:
A blank line before `static Gobjekt` would be nice.

Reply:
done